core/chains/evm/txmgr/models.go

```diff
@@ -18,19 +18,24 @@ import (
 
 	"github.com/smartcontractkit/chainlink/core/assets"
 	"github.com/smartcontractkit/chainlink/core/chains/evm/gas"
+	evmtypes "github.com/smartcontractkit/chainlink/core/chains/evm/types"
 	"github.com/smartcontractkit/chainlink/core/logger"
 	cnull "github.com/smartcontractkit/chainlink/core/null"
 	"github.com/smartcontractkit/chainlink/core/services/pg/datatypes"
 	"github.com/smartcontractkit/chainlink/core/utils"
 )
 
 // EthTxMeta contains fields of the transaction metadata
+// Not all fields are guaranteed to be present
 type EthTxMeta struct {
-	JobID int32 `json:"JobID"`
+	JobID *int32 `json:"JobID,omitempty"`
+
+	// Pipeline fields
+	FailOnRevert null.Bool `json:"FailOnRevert,omitempty"`
 
 	// VRF-only fields
-	RequestID     common.Hash `json:"RequestID,omitempty"`
-	RequestTxHash common.Hash `json:"RequestTxHash,omitempty"`
+	RequestID     *common.Hash `json:"RequestID,omitempty"`
+	RequestTxHash *common.Hash `json:"RequestTxHash,omitempty"`
 	// Batch variants of the above
```

Comment:
These were all made nullable fields since otherwise it fills the database with useless zero addresses and such in the JSON